arch/x86_64/inject-content.S

```diff
@@ -0,0 +1,82 @@
+#include "utils/asm.h"
+
+GLOBAL(inject_content)
+	/*
+	 * rsi is going to contain the address of free().
+	 * it's going to get wiped out by the call to malloc(),
+	 * so save it on the stack for later
+	 */
+	push %rsi
+	/*
+	 * same thing for rdx, which will contain the address
+	 * of _dl_open()
+	 */
+	push %rdx
+
+	/*
+	 * save previous value of r9,
+	 * because we're going to use it to call malloc()
+	 */
+	push %r9
+	// now move the address of malloc() into r9
+	mov %rdi,%r9
+	/*
+	 * choose the amount of memory to allocate with malloc()
+	 * based on the size of the path to the shared library
+	 * passed via rcx
+	 */
+	mov %rcx,%rdi
+	// now call r9; malloc()
+	callq *%r9
+	// after returning from malloc(),
+	// pop the previous value of r9 off the stack
+	pop %r9
+	// break in so that we can see what malloc() returned
+	int $3
+	// get the address of dlopen() off of
```

Comment:
Please add a blank line and some more comments so that it can be noticed it's executed in a separate phase.

Reply:
And please use `/* */` style comment instead of `//` to make the code look same as others.